pkg/crc/api/websocket/status.go

```diff
@@ -0,0 +1,52 @@
+package websocket
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/crc-org/crc/pkg/crc/logging"
+	"github.com/crc-org/crc/pkg/crc/machine"
+)
+
+type StatusConnectionListener struct {
+	machine machine.Client
+	done    chan bool
+}
+
+func NewStatusListener(machine machine.Client) ConnectionListener {
+	return &StatusConnectionListener{
+		machine: machine,
+		done:    make(chan bool),
+	}
+}
+
+func (s StatusConnectionListener) onClientConnected(count int, sendData SendData) {
+	if count == 1 {
```

Comment:
does it mean, only one ws `client` will be getting the status?

Reply:
No, that mean that collection of CPU/RAM usage start only when first client connected, but all connected clients will receive info.